test/python/test_qasm_parser.py

```diff
@@ -74,6 +79,29 @@ def test_parser_fail(self):
             QasmError, "Perhaps there is a missing", parse, file_path=self.qasm_file_path_fail
         )
 
+    def assert_invalid_version_failure(self, file):
+        """Helper method for ensuring invalid QASM versions throw an exception."""
+        self.assertRaisesRegex(
+            QasmError,
+            "Invalid version string: .+\\. This module supports OPENQASM 2.0; only.",
+            parse,
+            file_path=file,
+        )
```

Comment:
I know you were copying the other functions in this module (which is a good thing in general), but there's a clearer way of writing these that I'm slowly trying to transition the tests to when appropriate:
```python
with self.assertRaisesRegex(QasmError, "Invalid version string <...>"):
    parse(filename)
```

Personally, I'd not wrap this up in a separate function - it's very short code used only twice, and in tests it's not a bad thing to be a little verbose in order to see exactly what's being tested.  The way the tests are parameterised in this class is rather lacking (not your fault at all, but we absolutely should not need separate files here, and calculating the directory only during instance setup is a nuisance), but we could remove the need for this function by doing
```python
import ddt

@ddt.ddt
class TestParser:
    ...

    @ddt.data("example_version_fail.qasm", "example_minor_version_fail.qasm")
    def test_parser_version_fail(self, filename):
        filename = os.path.join(self.qasm_dir, filename)
        with self.assertRaisesRegex(QasmError, "Invalid version string <...>"):
            parse(filename)
```
Now it's clear that I'm running multiple tests of the same thing, and the parametrisation is explicitly at listed at the call site.

Reply:
Alright, got it, this is a useful lesson overall, thank you!